src/components/SearchInput/SearchInput.tsx

```diff
@@ -0,0 +1,121 @@
+import {
+  AriaAttributes,
+  ChangeEvent,
+  FocusEvent,
+  MouseEvent,
+  ReactNode,
+  useEffect,
+  useRef,
+  useState,
+} from "react"
+import styles from "./SearchInput.module.css"
+
+interface SearchInputProps extends Pick<AriaAttributes, "aria-label"> {
+  name?: string
+  placeholder?: string
+  searchIcon?: ReactNode
+  clearIcon?: ReactNode
+  debounceDelay?: number
+  value?: string
+  onSearch?: (searchTerm: string) => void
+  showSearchButton?: boolean
+  onBlur?: (e: FocusEvent<HTMLInputElement>) => void
+  onFocus?: (e: FocusEvent<HTMLInputElement>) => void
+}
+
+export function SearchInput({
+  name,
+  placeholder,
+  searchIcon,
+  clearIcon,
+  onSearch,
+  debounceDelay = 300,
+  showSearchButton,
+  value = "",
+  onBlur,
+  onFocus,
+  ...aria
+}: SearchInputProps) {
+  const [searchTerm, setSearchTerm] = useState("")
+  const [cancelButtonHidden, setCancelButtonHidden] = useState(true)
+  const timer = useRef<NodeJS.Timeout>()
+
+  useEffect(() => {
+    setSearchTerm(value)
+  }, [value])
+
+  useEffect(() => {
+    if (searchTerm && !showSearchButton) {
+      timer.current = setTimeout(() => onSearch?.(searchTerm), debounceDelay)
+    }
+    return () => {
+      clearTimeout(timer.current)
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [searchTerm, debounceDelay])
+
+  const handleReset = (e: MouseEvent<HTMLButtonElement>) => {
+    // we don't want the field to blur when a user clears the input
+    e.preventDefault()
+    setSearchTerm("")
+    onSearch?.("")
+    setCancelButtonHidden(true)
+  }
+
```

Comment:
One thing I noticed is we're missing submit handling via the Enter key. I think we can do this by adding:

```
useKeyPress(["Enter"], () => {
    onSearch?.(searchTerm)
  })
```

Reply:
Ah yeah, good catch.

Reply:
Do you think this should apply in both cases (i.e. with or without the search button)? I'm tempted to say it should apply conditionally on whether the button is present since the debounced callback is fired off anytime the user stops typing.

Reply:
I think it should apply whether or not the search button is present.

Reply:
>Seems like overkill for onSearch to fire after the timeout and additionally with Enter.

I would expect the timeout not to fire after `Enter` was pressed